### PR TITLE
add session_types api with proper relationships

### LINKED ISSUE #3717: Implement session_types api with proper relationships

### PR DESCRIPTION
**Pull request by shubham-padia**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
add tracks api with proper relationships

#### Changes proposed in this pull request:

- add session_types api with proper relationships
- 
-

Fixes #3717

### CODE REVIEW

File: app/api/__init__.py [hunk 1]
```diff
@@ -69,3 +73,9 @@
 api.route(TrackDetail, 'track_detail', '/tracks/<int:id>', '/sessions/<int:session_id>/track')
 api.route(TrackRelationship, 'track_sessions', '/tracks/<int:id>/relationships/sessions')
 api.route(TrackRelationship, 'track_event', '/tracks/<int:id>/relationships/event')
+
+# session_types
+api.route(SessionTypeList, 'session_type_list', '/session_types', '/events/<int:event_id>/session_types')
```
Comment (niranjan94) [inline app/api/__init__.py]: @shubham-padia this should exist only as a relationship to an event. `/session_types` shouldn't exist

Comment (shubham-padia) [inline app/api/__init__.py]: Changes done. The endpoints remaining are 
- /events/<int:id>/relationships/session_types
- /sessions/<int:id>/relationships/session_type
- /events/<int:event_id>/session_types
- /sessions/<int:session_id>/session_type
Is that correct?

Comment (shubham-padia) [inline app/api/__init__.py]: @niranjan94 :point_up_2: :point_up_2:

Comment (SaptakS) [inline app/api/__init__.py]: @shubham-padia @niranjan94 should we keep /session_types/<id> for update and deletion? I think we can keep that.

**Comment by shubham-padia** [comment]
@SaptakS Can you review this again, github dismisses review automatically after resolving merge conflicts

File: app/api/session_types.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    decorators = (jwt_required,)
+    schema = SessionTypeSchema
+    data_layer = {'session': db.session,
+                  'model': SessionType,
+                  'methods': {
+                      'before_get_object': before_get_object
+                  }}
+
+
```
Comment (SaptakS) [inline app/api/session_types.py]: Keep just one blank line at the end of a file. Thanks.

**Comment by SaptakS** [comment]
Why is there no relationship attribute to event or sessions in session_types.py? Not a review. Just curious. I see there is no SessionTypeRelationship

**Comment by shubham-padia** [comment]
@SaptakS that was to keep the 4 endpoints only in the earlier comments, as some more endpoints have to be added here following the discussion by niranjan , I'll make changes accordingly.

**Comment by shubham-padia** [comment]
@SaptakS changes done. Can u review this please?

File: app/api/session_types.py (hunk 2 above)
Comment (SaptakS) [inline app/api/session_types.py]: Resolved

**Pull request merged**